dynamic_rest/content_type_field.py

```diff
@@ -0,0 +1,73 @@
+import traceback
+
+from dynamic_rest.fields import DynamicField
+from dynamic_rest.routers import DynamicRouter
+
+
+class DynamicContentTypeField(DynamicField):
+
+    def __init__(
+        self,
+        object_field=None,
+        type_field=None,
+        id_field=None,
+        *args, **kwargs
+    ):
+        # NOTE: object_field should be same as `source`, except `source`
+        #       isn't reliably available until this field is bound().
+        if not object_field:
+            raise Exception(
+                'DynamicContentTypeField requires `object_field` kwarg'
+            )
+
+        # Infer type field and id field from object_field if not explicitly set
+        type_field = type_field or object_field + '_type'
+        id_field = id_field or object_field + '_id'
+
+        # Inject `requires` so required fields get prefetched properly.
+        # TODO: It seems like we should be able to require the type and
+        #       id fields, but that seems to conflict with some internal
+        #       Django magic. Disabling `.only()` by requiring '*' seem
+        #       to work more reliably...
```

Comment:
Interesting, nice call-out. I'm pleasantly surprised that `prefetch_related` is even compatible with `GenericForeignKey` :)